### ISSUE TITLE
Documentation for COleDateTime missing #include requirements

### ISSUE THREAD

**Issue by wtblock** [MicrosoftDocs/cpp-docs#4677]
[Enter feedback here]
Unless I missed it, I could not find the #include file necessary to use the COleDateTime class.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 26c31362-cb6f-cd6a-f0cb-b03a6f23cda0
* Version Independent ID: 64066ad0-5778-4f58-9d0e-7ef1978f9f86
* Content: [COleDateTime Class](https://learn.microsoft.com/en-us/cpp/atl-mfc-shared/reference/coledatetime-class?view=msvc-150&devlangs=cpp&f1url=%3FappId%3DDev15IDEF1%26l%3DEN-US%26k%3Dk(COleDateTime)%3Bk(DevLang-C%252B%252B)%3Bk(TargetOS-Windows)%26rd%3Dtrue#setstatus)
* Content Source: [docs/atl-mfc-shared/reference/coledatetime-class.md](https://github.com/MicrosoftDocs/cpp-docs/blob/main/docs/atl-mfc-shared/reference/coledatetime-class.md)
* Product: **visual-cpp**
* Technology: **cpp-mfc**
* GitHub Login: @TylerMSFT
* Microsoft Alias: **twhitney**

**Comment by orcmid** [comment]
`ATLComTime.h`

Look way down the page for "Requirements."  A number of API descriptions have their `#include` requirements buried down at the end in that manner.

**Comment by wtblock** [comment]
I searched the entire article for "#include" and there were no "#includes" in the article. The word "include" shows up twice but is unrelated to the C++ include.

Is it possible for the feedback linked to the wrong page?

This is the page I'm referring to, which is where the F1 key took me when I needed help in Visual Studio 2017:

https://learn.microsoft.com/en-us/cpp/atl-mfc-shared/reference/coledatetime-class?view=msvc-150

[signatureImage]
________________________________

**Comment by orcmid** [comment]
No, it is there.   It is not with a `#include`.  There is a title **Requirements** and a statement

   Header: `ATLComTime.h`

It is not in an obvious place.  There's much verbiage before it gets to that point and then the individual functions and examples.

I agree that it should have more-prominent placement and consistency among the **Requirements** statements be pursued.

**Comment by wtblock** [comment]
Thank you very much Dennis.

I don’t know why I did not think to search for “header” instead of "include.”

By the way, I have found the new Bing AI to be invaluable for finding this type of information while coding, but its response was not as precise in that it recommended wider ranging include.

My question was botched by the voice to text to look like this:

What C++ header is required to use the class C OLE Date time?

The Bing AI response figured out what I was asking:

To use the class COleDateTime in C++, you need to include the header file **afxdisp.h**¹². This header file contains the declarations for the MFC (Microsoft Foundation Class) library of C++, which provides various classes and functions for working with date and time values, OLE automation, and other features¹. COleDateTime is one of the possible types for the VARIANT data type of OLE automation, which can store different kinds of data in a single variable¹. COleDateTime encapsulates the DATE data type that is used in OLE automation, and provides methods for converting, formatting, and working with date and time values¹.

For example, you can create a COleDateTime object with a specific date and time value like this:

`COleDateTime myDate (2023, 8, 11, 11, 24, 7); // 11 August 2023, 11:24:07 AM`

And you can format it as a string using the Format method:

`CString strDate = myDate.Format (_T ("%Y-%m-%d %H:%M:%S")); // strDate contains "2023-08-11 11:24:07"`

I hope this answers your question. Do you have any other questions about C++ or COleDateTime? 😊

Source: Conversation with Bing, 8/11/2023
(1) COleDateTime Class | Microsoft Learn. https://learn.microsoft.com/en-us/cpp/atl-mfc-shared/reference/coledatetime-class?view=msvc-170.
(2) C++ (Cpp) COleDateTime Examples. https://cpp.hotexamples.com/examples/-/COleDateTime/-/cpp-coledatetime-class-examples.html.
(3) Date and Time | Microsoft Learn. https://learn.microsoft.com/en-us/cpp/atl-mfc-shared/date-and-time?view=msvc-170.
(4) c++ - COleDateTime ambigous symbol - Stack Overflow. https://stackoverflow.com/questions/34762289/coledatetime-ambigous-symbol.

***@***.***

**Comment by TylerMSFT** [comment]
Yes, the file to include is listed way at the bottom. It's not ideal.
Unfortunately, we have accrued more issues than our resources can manage. Criteria for closing include severity of the issue, how broadly applicable it is, etc. Based on that, I'm closing this one out. Sorry :-(

**Comment by orcmid** [comment]
@TylerMSFT @wtblock 

I nosed around and the use of **Requirements** with a naming of headers seems to be consistent throughout the ATL/MFC materials.  There's a consistent pattern in the sequence of subsections.

I sympathize over the triage decision.

Although I've learned how to edit the MicrosoftDocs material, this is too distant from my area of interest as well as too much change for little gain.  That may even be true where I am more engaged :(.
